### ISSUE TITLE
`for_each` value is not iterable with sensitive values in ternary condition

### ISSUE THREAD

**Issue by bodgit** [terraform-linters/tflint#2345]
### Summary

All tflint versions since 0.55.0 have started erroring on this particular `for_each` condition. The Terraform is valid and plans and applies correctly. Downgrading tflint below 0.55.0 stops the error.

### Command

tflint

### Terraform Configuration

```terraform
variable “username” {
  type      = string
  sensitive = true
  default   = null
}

variable “password” {
  type      = string
  sensitive = true
  default   = null
}

variable “agents” {
  type = map(object({
    ip_address      = string
    server_hostname = string
    subdirectory    = string
    active          = optional(bool, false)
  }))
  default = {}
}

resource “aws_datasync_location_smb” “test” {
  for_each = var.username != null && var.password != null ? var.agents : {}
  …
}
```

### TFLint Configuration

```HCL
plugin “terraform” {
  enabled = true
  preset  = “all”
}

plugin “aws” {
  enabled = true
  version = “0.40.0”
  source  = “github.com/terraform-linters/tflint-ruleset-aws”
}
```

### Output

```Shell
Failed to check ruleset; failed to check “aws_iam_policy_sid_invalid_character” rule: efs.tf:295,21-110: The `for_each` value is not iterable; `cty.MapValEmpty(cty.Object(map[string]cty.Type{“active”:cty.Bool, “ip_address”:cty.String, “server_hostname”:cty.String, “subdirectory”:cty.String})).Mark(marks.Sensitive)` is not iterable, and 1 other diagnostic(s)
```

### TFLint Version

0.55.0

### Terraform Version

1.5.7

### Operating System

- [x] Linux
- [ ] macOS
- [ ] Windows

**Comment by bodgit** [comment]
Disabling the sensitive attribute on the two string variables seems to fix the error but I don't really want to do that as they're AD credentials.

**Comment by bodgit** [comment]
Judicious sprinkling of `nonsensitive()` in the `for_each` condition seems to keep tflint happy, i.e:
```hcl
for_each = nonsensitive(var.username) != null && nonsensitive(var.password) != null ? var.agents : {}
```
